zachsinterview/src/main/java/com/imprev/zachsinterview/NumberStoreAdditionalOperationsImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.imprev.zachsinterview;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class NumberStoreAdditionalOperationsImpl implements NumberStoreAdditionalOperations {
+    
+    private NumberStore numberStore;
+    
+    public NumberStoreAdditionalOperationsImpl(NumberStore numberStore) {
+        this.numberStore = numberStore;
+    }
+
+    @Override
+    public Integer sumByPrefix(String prefix) {
+        
+        int sum = 0;
+        try {
+            Iterator<String> iter = this.numberStore.getKeysByPrefix(prefix);
+            List<String> allKeys = new ArrayList<String>();
+            iter.forEachRemaining(allKeys::add);
+            
+            for (String key : allKeys) {
+                sum += this.numberStore.get(key);
+            }
+            
+        } catch (NumberStoreException e) {
+            e.printStackTrace();
```

Comment:
I think we want to throw an exception here, instead of returning a potential partial sum (i..e item 4 of 1000 fails, returning a sum of the first 3 might give the impression that the operation was successful)